packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -853,6 +854,12 @@ export class GridPanel<T = {}> extends PureComponent<Props<T>, State> {
                 // update/set sorts and filters to combine view and user-defined items
                 filters: model.filterArray.concat(view.filters),
                 sorts: model.sorts.concat(view.sorts),
+                columns: model.displayColumns.map(col => {
```

Comment:
nit: This column logic of translating query columns to view info columns is repeated several times. Consider consolidating into a utility:

```tsx
// Note: You will need to export `ViewInfoColumn` from `ViewInfo.ts`
viewColumn = (column: QueryColumn): ViewInfoColumn => {
    return {
        // Issue 46256: use encoded fieldKeyPath
        fieldKey: column.fieldKeyPath,
        title: column.customViewTitle,
    };
};
```

Then the subsequent usages can make use of this:

```tsx
hideColumn = (columnToHide: QueryColumn): void => {
    const { model } = this.props;
    this.saveAsSessionView({
        columns: model.displayColumns
            .filter(column => column.index !== columnToHide.index)
            .map(this.viewColumn),
    });
};

updateColumnTitle = (updatedCol: QueryColumn): void => {
    const { model } = this.props;
    this.saveAsSessionView({
        columns: model.displayColumns.map(col => (
            this.viewColumn(col.index === updatedCol.index ? updatedCol : col)
        )),
    });
    this.setState({ disableColumnDrag: false });
};

onColumnDrop = (source: string, target: string): void => {
    const { displayColumns } = this.props.model;

    const sourceIndex = displayColumns.findIndex(col => col.index === source);
    const colInMotion = displayColumns.find(col => col.index === source);
    if (colInMotion) {
        let updatedColumns = displayColumns.filter(col => col.index !== source);
        const targetIndex = updatedColumns.findIndex(col => col.index === target);
        if (targetIndex > -1 && targetIndex !== sourceIndex) {
            updatedColumns = [
                ...updatedColumns.slice(0, targetIndex),
                colInMotion,
                ...updatedColumns.slice(targetIndex),
            ];

            this.saveAsSessionView({ columns: updatedColumns.map(this.viewColumn) });

            incrementClientSideMetricCount('customViews', 'columnReorderDragNDrop');
        }
    }
};
```